plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/function/tablechanges/DeltaLakeTableChangesFunction.java

```diff
@@ -0,0 +1,190 @@
+        this.statisticsAccess = requireNonNull(statisticsAccess, "statisticsAccess is null");
+        this.fileSystemFactory = requireNonNull(fileSystemFactory, "fileSystemFactory is null");
+    }
+
+    @Override
+    public TableFunctionAnalysis analyze(ConnectorSession session, CatalogHandle catalogHandle, ConnectorTransactionHandle transaction, Map<String, Argument> arguments)
+    {
+        ScalarArgument tableNameArgument = (ScalarArgument) arguments.get(TABLE_NAME_ARGUMENT);
+        requireNonNull(tableNameArgument.getValue(), "tableName value for function table_changes() is null");
+        String schemaTableName = ((Slice) tableNameArgument.getValue()).toStringUtf8();
+        if (!schemaTableName.contains(".") || schemaTableName.indexOf(".") != schemaTableName.lastIndexOf(".")) {
+            throw new IllegalArgumentException("table_name value for function table_changes() must be in following format: schema_name.table_name");
+        }
+
+        ScalarArgument startVersionArgument = (ScalarArgument) arguments.get(START_VERSION_ARGUMENT);
+        Long startVersion = (Long) startVersionArgument.getValue();
+        requireNonNull(startVersion, "startVersion value for function table_changes() is null");
+        startVersion++; // to guarantee exclusiveness for the lower end of the range
+        checkArgument(startVersion >= 0, "startVersion value for function table_changes() must not be negative");
+
+        ScalarArgument endVersionArgument = (ScalarArgument) arguments.get(END_VERSION_ARGUMENT);
+        Long endVersion = (Long) endVersionArgument.getValue();
+        requireNonNull(endVersion, "endVersion value for function table_changes() is null");
+        checkArgument(endVersion >= 0, "endVersion value for function table_changes() must not be negative");
+        checkArgument(endVersion >= startVersion, "endVersion value for function table_changes() must be greater than startVersion");
+
+        int indexOfDot = schemaTableName.indexOf(".");
+        String schemaName = schemaTableName.substring(0, indexOfDot);
+        String tableName = schemaTableName.substring(indexOfDot + 1);
+        MetadataEntry metadata = getTableMetadata(session, schemaName, tableName);
+
+        ImmutableList.Builder<Descriptor.Field> outputFields = ImmutableList.builder();
+        List<DeltaLakeColumnMetadata> deltaLakeColumnMetadata = extractSchema(metadata, typeManager);
+        deltaLakeColumnMetadata.stream()
+                .map(column -> new Descriptor.Field(column.getName(), Optional.of(column.getType())))
+                .forEach(outputFields::add);
+        outputFields.add(new Descriptor.Field(CHANGE_TYPE_COLUMN_NAME, Optional.of(VARCHAR)));
+        outputFields.add(new Descriptor.Field(COMMIT_VERSION_COLUMN_NAME, Optional.of(INTEGER)));
+        outputFields.add(new Descriptor.Field(COMMIT_TIMESTAMP_COLUMN_NAME, Optional.of(TIMESTAMP_TZ_MILLIS)));
+        List<Descriptor.Field> requiredColumns = outputFields.build();
```

Comment:
What are we doing for the tables containing CDF column names?
We should probably throw an exception right?

https://github.com/delta-io/delta/blob/e0e9b0095dcc5b1c4372474a54c87428340fe899/core/src/test/scala/org/apache/spark/sql/delta/MergeIntoSQLSuite.scala#L353-L366

This may be relevant for #16221

Reply:
Also for `CREATE TABLE` with `delta.enableChangeDataFeed` this should be relevant.

Reply:
We can do it as a follow up